new_samples/worker/worker.go

```diff
@@ -0,0 +1,86 @@
+// Package worker implements a Cadence worker with basic configurations.
+package worker
+
+import (
+	"github.com/uber-common/cadence-samples/new_samples/workflows"
+	"github.com/uber-go/tally"
+	apiv1 "github.com/uber/cadence-idl/go/proto/api/v1"
+	"go.uber.org/cadence/.gen/go/cadence/workflowserviceclient"
+	"go.uber.org/cadence/activity"
+	"go.uber.org/cadence/compatibility"
+	"go.uber.org/cadence/worker"
+	"go.uber.org/cadence/workflow"
+	"go.uber.org/yarpc"
+	"go.uber.org/yarpc/transport/grpc"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+const (
+	HostPort       = "127.0.0.1:7833"
+	Domain         = "cadence-samples-domain"
+	TaskListName   = "cadence-samples-worker"
+	ClientName     = "cadence-samples-worker"
+	CadenceService = "cadence-frontend"
+)
+
+// StartWorker creates and starts a basic Cadence worker.
+func StartWorker() {
+	// TaskListName identifies set of client workflows, activities, and workers.
+	// It could be your group or client or application name.
+	logger, cadenceClient := buildLogger(), buildCadenceClient()
+	workerOptions := worker.Options{
+		Logger:       logger,
+		MetricsScope: tally.NewTestScope(TaskListName, map[string]string{}),
+	}
+
+	w := worker.New(
+		cadenceClient,
+		Domain,
+		TaskListName,
+		workerOptions)
+	// HelloWorld workflow registration
+	w.RegisterWorkflowWithOptions(workflows.HelloWorldWorkflow, workflow.RegisterOptions{Name: "cadence_samples.HelloWorldWorkflow"})
+	w.RegisterActivityWithOptions(workflows.HelloWorldActivity, activity.RegisterOptions{Name: "cadence_samples.HelloWorldActivity"})
+
+	err := w.Start()
+	if err != nil {
+		panic("Failed to start worker")
+	}
+	logger.Info("Started Worker.", zap.String("worker", TaskListName))
+
+}
+
+func buildCadenceClient() workflowserviceclient.Interface {
```

Comment:
tbh I'm surprised this is so simple.  neat.  and thanks!

Reply:
yeah spent a lot of time trying to find the easiest one. I believe this will definitely try to raise our adoption rate.